### PR TITLE
update to msal4.30 + pkce support

### PR DESCRIPTION
**Pull request by jennyf19**
@jmprieur have only tested sign-in (obviously, to start), but thought you might be interested as well.

#470

### CODE REVIEW

File: src/Microsoft.Identity.Web/TokenAcquisition.cs [hunk 1]
```diff
@@ -151,12 +152,14 @@ internal partial class TokenAcquisition : ITokenAcquisitionInternal
             {
                 var application = GetOrBuildConfidentialClientApplication();
 
+                context.Properties.Items.TryGetValue(OAuthConstants.CodeVerifierKey, out string? codeVerifier);
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web/TokenAcquisition.cs:155]: might move this inline, but leaving as-is for testing.

File: src/Microsoft.Identity.Web/TokenAcquisition.cs [hunk 2]
```diff
@@ -151,12 +152,14 @@ internal partial class TokenAcquisition : ITokenAcquisitionInternal
             {
                 var application = GetOrBuildConfidentialClientApplication();
 
+                context.TokenEndpointRequest.Parameters.TryGetValue(OAuthConstants.CodeVerifierKey, out string? codeVerifier);
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web/TokenAcquisition.cs:155]: can move this inline, but left for testing.

**Review by jmprieur — COMMENTED** [review]
LTGM @jennyf19 
I also tried by overriding `usePkce = false` in the OpenIdConnectOptions;

**Review by jmprieur — CHANGES_REQUESTED** [review]
We shouldn't remove line 434 of  src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilderExtensions.cs

otherwise 

.AddMicrosoftIdentityWebApp()

and then

```CSharp
services.ConfigureOptions<MicrosoftIdentityOptions>(options =>
{
 options.UsePkce = false;
});
```

would not do the right thing?

**Comment by bgavrilMS** [comment]
How do you know that this works?

**Comment by jmprieur** [comment]
@bgavrilMS : you run the WebAppCallsWebApi projects and use the todo-list button to call the web API.
When you sign-in, this is used (By default PKCE is enabled in ASP.NET Core 5.0); 

The other thing to test is that if you override OpenIdConnectOptions.UsePkce to false, this still works (and it does)

**Comment by jennyf19** [comment]
@bgavrilMS we discussed thursday, the testing would have to happen in MSAL.NET, which I think you added. I tested it manually, as we discussed, i put a break point and checked the code verifier was included and also checked when it's not. everything would break if it didn't work. I'm not testing cca.GetAuthorizationRequestUrl, as we don't use it.

File: src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilderExtensions.cs [hunk 3]
```diff
@@ -431,7 +431,6 @@ internal static void PopulateOpenIdOptionsFromMicrosoftIdentityOptions(OpenIdCon
             options.SkipUnrecognizedRequests = microsoftIdentityOptions.SkipUnrecognizedRequests;
             options.DisableTelemetry = microsoftIdentityOptions.DisableTelemetry;
             options.NonceCookie = microsoftIdentityOptions.NonceCookie;
-            options.UsePkce = microsoftIdentityOptions.UsePkce;
 #if DOTNET_50_AND_ABOVE
```
[the remark this answers is not in the document]
Comment (jennyf19) [reply src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilderExtensions.cs]: yes. i re-added it.

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @jennyf19

**Comment by bgavrilMS** [comment]
@jennyf19 - you have an E2E / Selenium test I believe. Will it use PKCE now? 
I am asking if the team needs to manually validate this scenario on every release or if it is covered by existing automation.

**Comment by jennyf19** [comment]
@bgavrilMS I don't understand what the issue is...? We use PKCE now always, so it just works. Not sure what you are asking.

**Comment by tedvanderveen** [comment]
@jennyf19 great work, can we pick it up in a nightly build package or should we wait for next Nuget release?

**Comment by jennyf19** [comment]
@tedvanderveen i've been focused on getting an enhancement out for handling multiple auth schemes....I will sync with @jmprieur and maybe we'll do a release with this and a few other completed items in [1.10 project board](https://github.com/AzureAD/microsoft-identity-web/projects/24) so we aren't holding things up for the larger enhancement work with auth schemes.

**Comment by jennyf19** [comment]
@tedvanderveen actually, follow [1.9.2](https://github.com/AzureAD/microsoft-identity-web/projects/25), which will have the PCKE included. We will try to have it out next week (1st week of May).

**Comment by tedvanderveen** [comment]
@jennyf19 looks like 1.9.2 is ready to roll. Great work!

**Comment by jennyf19** [comment]
@tedvanderveen 1.9.2 is out. should be on nuget shortly: Included in [1.9.2 release](https://github.com/AzureAD/microsoft-identity-web/releases/tag/1.9.2)

**Pull request merged**